packages/main/src/CalendarHeader.js

```diff
@@ -154,6 +185,51 @@ class CalendarHeader extends UI5Element {
 		}
 	}
 
+	_getDisplayedSecondaryMonthText(month) {
+		const localeData = getCachedLocaleDataInstance(getLocale());
+		const pattern = localeData.getIntervalPattern();
+		const secondaryMonthsNames = getCachedLocaleDataInstance(getLocale()).getMonthsStandAlone("abbreviated", this.secondaryCalendarDate);
+		const secondaryMonthsNamesWide = getCachedLocaleDataInstance(getLocale()).getMonthsStandAlone("wide", this.secondaryCalendarDate);
+
+		if (month.startMonth === month.endMonth) {
+			return {
+				text: localeData.getMonths("abbreviated", this.secondaryCalendarType)[month.startMonth],
+				textInfo: localeData.getMonths("wide", this.secondaryCalendarType)[month.startMonth],
+			};
+		}
+
+		return {
+			text: pattern.replace(/\{0\}/, secondaryMonthsNames[month.startMonth]).replace(/\{1\}/, secondaryMonthsNames[month.endMonth]),
+			textInfo: pattern.replace(/\{0\}/, secondaryMonthsNamesWide[month.startMonth]).replace(/\{1\}/, secondaryMonthsNamesWide[month.endMonth]),
+		};
+	}
+
+	_getDisplayedSecondaryMonths(localDate) {
+		let firstDate = CalendarDate.fromLocalJSDate(localDate);
+		firstDate.setDate(1);
+		firstDate = new CalendarDate(firstDate, this.secondaryCalendarType);
+
+		const startMonth = firstDate.getMonth();
+		let lastDate = CalendarDate.fromLocalJSDate(localDate, this.primaryCalendarType);
+
+		lastDate.setDate(this._daysInMonth(lastDate));
+		lastDate = new CalendarDate(lastDate, this.secondaryCalendarType);
+		const endMonth = lastDate.getMonth();
+		return { startMonth, endMonth };
+	}
+
+	_daysInMonth(calendarDate) {
+		calendarDate = new CalendarDate(calendarDate);
+		calendarDate.setDate(1);
+		calendarDate.setMonth(calendarDate.getMonth() + 1);
+		calendarDate.setDate(0);
```

Comment:
Do you really need to set the date to 1 and then to 0

Reply:
the short answer is yes :) this logic prevents this case: if we have a date 31.01.2020 and then set the month +1 as a result we will have 02.03.2020

Reply:
_daysInMonth is perfect candidate for a static method in an utility module like the ExtremeDates module,
but not sure if we should do it via the current change.